### PR TITLE
Cloud Assets - build and push docker command

### PR DESCRIPTION
**Pull request by jhorvath**
Added a command to build and push the Docker image to a selected repository. I cleaned up the dependencies of maven.indexer. Note that javax.inject is now in a separate module within the platform cluster.

### CODE REVIEW

File: enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/ImageBuilderCommand.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+                            FileObject pomFile = projectDirectory.getFileObject("pom.xml");
+                            if (pomFile != null) {
+                                pomFile.refresh();
+                            }
+                            
+                            nbMavenProject.getFreshProject().thenAccept(mvnProject -> {
+                                List<String> goals;
+                                String version = mvnProject.getVersion();
+                                if (isGdk) {
+                                    goals = List.of("compile", "deploy", "-pl", "oci", "-Dpackaging=docker", "-Djib.to.image=" + repository.getUrl()+ ":" + version);
```
Comment (sdedic) [inline enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/ImageBuilderCommand.java:145]: I think this if-else can be replaced by `RunConfig.setReactorStyle(ALSO_MAKE)` that generates `--projects <list> --also-make` into the commandline.

File: enterprise/websvc.restlib/nbproject/project.xml [hunk 2]
```diff
@@ -50,12 +50,21 @@
...
+                        <specification-version>2.61</specification-version>
+                    </run-dependency>
+                </dependency>
             </module-dependencies>
             <public-packages>
                 <package>jakarta.annotation</package>
                 <package>jakarta.annotation.security</package>
                 <package>jakarta.annotation.sql</package>
-                <package>javax.inject</package>
+                <!--<package>javax.inject</package>-->
```
Comment (sdedic) [inline enterprise/websvc.restlib/nbproject/project.xml:67]: Leftover comment ?

File: java/maven.embedder/nbproject/project.xml [hunk 3]
```diff
@@ -326,9 +333,9 @@
             <class-path-extension>
                 <runtime-relative-path>../maven/lib/javax.annotation-api-1.3.2.jar</runtime-relative-path>
             </class-path-extension>
-            <class-path-extension>
+<!--            <class-path-extension>
```
Comment (sdedic) [inline java/maven.embedder/nbproject/project.xml:336]: leftover comment

File: platform/libs.javax.inject/external/jakarta.inject-2.6.1-license.txt [hunk 4]
```diff
@@ -0,0 +1,93 @@
+Name: HK2
+Version: 2.6.1
+Description: A light-weight and dynamic dependency injection framework
+License: EPL-v20
+Origin: https://github.com/eclipse-ee4j/glassfish-hk2
```
Comment (sdedic) [inline platform/libs.javax.inject/external/jakarta.inject-2.6.1-license.txt:5]: The name and origin seem incorrect

File: platform/libs.javax.inject/src/org/netbeans/libs/javax/Dummy.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.netbeans.libs.jsr223;
```
Comment (sdedic) [inline platform/libs.javax.inject/src/org/netbeans/libs/javax/Dummy.java:20]: bad package !

**Review by sdedic — CHANGES_REQUESTED** [review]

File: enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/ImageBuilderCommand.java [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+                            Files.write(tempFile, init.getBytes(StandardCharsets.UTF_8));
+                            RunConfig runConfig = RunUtils.createRunConfig(
+                                    project,
+                                    "",
+                                    "Build container image",
+                                    Collections.emptySet(),
+                                    "--init-script", tempFile.toAbsolutePath().toString(), "dockerBuild", "dockerPush"
+                            );
+                            ExecutorTask task = RunUtils.executeGradle(runConfig, "");
+                            task.waitFinished();
```
[the remark this answers is not in the document]
Comment (jhorvath) [reply enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/ImageBuilderCommand.java]: Temporary file is deleted in the TaskListener now

File: enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/ImageBuilderCommand.java [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+                                    goals = List.of("compile", "deploy", "-Dpackaging=docker", "-Djib.to.image=" + repository.getUrl()+ ":" + version);
+                                }
+                                org.netbeans.modules.maven.api.execute.RunConfig runConfig = org.netbeans.modules.maven.api.execute.RunUtils.createRunConfig(
+                                        FileUtil.toFile(project.getProjectDirectory()),
+                                        project,
+                                        "Build container image",
+                                        goals
+                                );
+                                ExecutorTask task = org.netbeans.modules.maven.api.execute.RunUtils.executeMaven(runConfig);
+                                task.waitFinished();
```
[the remark this answers is not in the document]
Comment (jhorvath) [reply enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/ImageBuilderCommand.java]: Using `TaskListener` now

File: enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/ImageBuilderCommand.java [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+                                return null;
+                            });
+                        });
+                    }
+                });
+        return result;
+    }
+
+    private CompletableFuture<String> confirmVersion(String version) {
+        NotifyDescriptor.InputLine desc = new NotifyDescriptor.InputLine("Enter image tag", "Enter an image tag");
```
Comment (sdedic) [inline enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/ImageBuilderCommand.java:201]: Nitpick: I18N

**Review by sdedic — APPROVED** [review]
Seems OK. 
As this adds a new module, I'll re-run the tests with `ci:all` so everything is built / tested

**Pull request merged**